### PR TITLE
CSHARP-906 Support dse auth transitional mode by default

### PR DESCRIPTION
**Pull request by joao-r-reis**

### CODE REVIEW

File: src/Cassandra.IntegrationTests/Core/TransitionalAuthenticationTests.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                    300);
+            }
+        }
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            Diagnostics.CassandraTraceSwitch.Level = TraceLevel.Info;
+            _testClusterForAuthTesting = GetTestCcmClusterForAuthTests();
+            RetryUntilClusterAuthHealthy(_testClusterForAuthTesting);
```
Comment (jorgebay) [inline src/Cassandra.IntegrationTests/Core/TransitionalAuthenticationTests.cs:64]: Is this related to a failure? I didn't get it on the Node.js driver with `-Dcassandra.superuser_setup_delay_ms=0`.

File: src/Cassandra.IntegrationTests/Core/TransitionalAuthenticationTests.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+                                 .Build())
+            {
+                var session = cluster.Connect();
+                var rs = session.Execute("SELECT * FROM system.local");
+                Assert.Greater(rs.Count(), 0);
+            }
+        }
+
+        [Test]
+        public void WrongCredentials_DsePlainText_AuthSuccess()
```
Comment (jorgebay) [inline src/Cassandra.IntegrationTests/Core/TransitionalAuthenticationTests.cs:127]: TIL that with transitional mode any credentials would do!

**Review by jorgebay — APPROVED** [review]
lgtm!

File: src/Cassandra.IntegrationTests/Core/TransitionalAuthenticationTests.cs (hunk 1 above)
Comment (joao-r-reis) [reply src/Cassandra.IntegrationTests/Core/TransitionalAuthenticationTests.cs:64]: Tbh I'm not sure why this is here, I copied this from the pre exiting authentication tests... I can give it a go and just remove this

Comment (jorgebay) [reply src/Cassandra.IntegrationTests/Core/TransitionalAuthenticationTests.cs:64]: bah, whatever you prefer, I think this was needed before "delay" setting to 0ms but its not anymore.

Comment (joao-r-reis) [reply src/Cassandra.IntegrationTests/Core/TransitionalAuthenticationTests.cs]: It seems that it wasn't needed indeed, tests are green without it :+1:

**Pull request merged**